src/lib/message/L1ToL2MessageCreator.ts

```diff
@@ -13,6 +13,43 @@ import { BigNumber } from 'ethers'
 import { SignerProviderUtils } from '../dataEntities/signerOrProvider'
 import { MissingProviderArbSdkError } from '../dataEntities/errors'
 import { getBaseFee } from '../utils/lib'
+import {
+  isL1ToL2TransactionRequest,
+  L1ToL2TransactionRequest,
+} from '../dataEntities/transactionRequest'
+
+interface L1ToL2MessageParams {
```

Comment:
this seems like a mix of the params from `IRetryableData` and `L1ToL2TransactionRequest` from `src/lib/dataEntities/transactionRequest.ts`.
I don't mind the repeated code, but maybe there's a neat way to conciliate these 3 interfaces to share more stuff

Reply:
updated to be a function of retryable data, now a breaking change